fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1484,16 +1654,181 @@ ACTOR static Future<bool> rebalanceTeams(DDQueueData* self,
 	    ShardsAffectedByTeamFailure::Team(sourceTeam->getServerIDs(), primary));
 	for (int i = 0; i < shards.size(); i++) {
 		if (moveShard == shards[i]) {
-			traceEvent->detail("ShardStillPresent", true);
-			self->output.send(RelocateShard(moveShard, priority));
+			self->output.send(RelocateShard(moveShard, priority, RelocateReason::REBALANCE_DISK));
 			return true;
 		}
 	}
 
-	traceEvent->detail("ShardStillPresent", false);
+	traceEvent->detail("SkipReason", "ShardNotPresent");
 	return false;
 }
 
+ACTOR Future<Void> getSrcDestTeams(DDQueueData* self,
+                                   int teamCollectionIndex,
+                                   GetTeamRequest srcReq,
+                                   GetTeamRequest destReq,
+                                   Reference<IDataDistributionTeam>* sourceTeam,
+                                   Reference<IDataDistributionTeam>* destTeam,
```

Comment:
The types of these parameters don't need to be pointers, because the `Reference`s are cheap to copy. Passing `Reference<T>*` parameters to actors can lead to some difficult to debug illegal memory access bugs if this actor outlives the lifetime of the referenced object.

Reply:
I think the reason I use pointers here rather than `Reference<IDataDistributionTeam> sourceTeam` is `sourceTeam = newReferenceVar;` won't change the underlying `ptr` member in the caller. What I want is a `Reference<T>&`, but ACTOR is not friendly for reference arguments, so...

Reply:
Is there still a chance of the memory access bug @sfc-gh-tclinkenbeard mentioned? Would it be better to return sourceTeam and destTeam as std::pair<Reference<IDataDistributionTeam>,Reference<IDataDistributionTeam>>?